core/src/main/java/com/redhat/lightblue/client/request/BulkLightblueDataRequest.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * 
+ */
+package com.redhat.lightblue.client.request;
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.client.request.data.DataDeleteRequest;
+import com.redhat.lightblue.client.request.data.DataFindRequest;
+import com.redhat.lightblue.client.request.data.DataInsertRequest;
+import com.redhat.lightblue.client.request.data.DataSaveRequest;
+import com.redhat.lightblue.client.request.data.DataUpdateRequest;
+
+/**
+ * @author bvulaj
+ *
+ */
+public class BulkLightblueDataRequest extends AbstractBulkLightblueRequest<AbstractLightblueDataRequest> {
+
+	public BulkLightblueDataRequest() {
+		super();
+	}
+
+	public BulkLightblueDataRequest(List<AbstractLightblueDataRequest> requests) {
+		super(requests);
+	}
+
+	@Override
+	public JsonNode getBodyJson() {
+		ObjectNode root = JsonNodeFactory.instance.objectNode();
+		ArrayNode reqs = JsonNodeFactory.instance.arrayNode();
+		for (AbstractLightblueDataRequest req : requests) {
+			ObjectNode seqNode = JsonNodeFactory.instance.objectNode();
+			seqNode.set("seq", JsonNodeFactory.instance.numberNode(reqs.size()));
+			String op = "";
+			if (req instanceof DataDeleteRequest) {
+				op = "delete";
+			} else if (req instanceof DataFindRequest) {
+				op = "find";
+			} else if (req instanceof DataInsertRequest) {
+				op = "insert";
+			} else if (req instanceof DataSaveRequest) {
+				op = "save";
+			} else if (req instanceof DataUpdateRequest) {
+				op = "update";
+			}
```

Comment:
Don't include classes from core.

enum is fine.

On Wed, Sep 9, 2015 at 8:05 AM, Brandon Vulaj <notifications@github.com>
wrote:

> In
> core/src/main/java/com/redhat/lightblue/client/request/BulkLightblueDataRequest.java
> <https://github.com/lightblue-platform/lightblue-client/pull/156#discussion_r39045932>
> :
>
> > +		ArrayNode reqs = JsonNodeFactory.instance.arrayNode();
> > +		for (AbstractLightblueDataRequest req : requests) {
> > +			ObjectNode seqNode = JsonNodeFactory.instance.objectNode();
> > +			seqNode.set("seq", JsonNodeFactory.instance.numberNode(reqs.size()));
> > +			String op = "";
> > +			if (req instanceof DataDeleteRequest) {
> > +				op = "delete";
> > +			} else if (req instanceof DataFindRequest) {
> > +				op = "find";
> > +			} else if (req instanceof DataInsertRequest) {
> > +				op = "insert";
> > +			} else if (req instanceof DataSaveRequest) {
> > +				op = "save";
> > +			} else if (req instanceof DataUpdateRequest) {
> > +				op = "update";
> > +			}
>
> I'll work that into this pull request. I prefer an enum for this case (we
> have CRUDOperations in core), but if either of you prefer constants
> instead, just shout.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/156/files#r39045932>
> .
>